superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx

```diff
@@ -335,7 +335,13 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
       return;
     }
 
-    // Case 2: Handle the default to first Value case
+    if (filterState.value !== undefined) {
```

Comment:
Are other nullish values something we handle differently? (like for resetting the value?)

Reply:
I think so this is reintroducing the logic as it was